### PR TITLE
added rpc to get team information about teams you are not in

### PR DESCRIPTION
**Pull request by AMarcedone**

### CODE REVIEW

File: go/teams/teams.go [hunk 1]
```diff
@@ -2707,3 +2707,75 @@ func KeySummary(t Teamer) string {
...
+	PublicMembers []struct {
+		Role     keybase1.TeamRole
+		UID      keybase1.UID
+		Username string
+	} `json:"public_members"`
+}
+
+var _ libkb.APIResponseWrapper = (*TeamInfo)(nil)
+
+func (r *TeamInfo) GetAppStatus() *libkb.AppStatus {
```
Comment (zapu) [inline go/teams/teams.go]: Use `AppStatusEmbed` like so:
```
	type resType struct {
		libkb.AppStatusEmbed
		...
	}
```

File: protocol/avdl/keybase1/teams.avdl [hunk 2]
```diff
@@ -185,6 +185,25 @@ protocol teams {
...
   }
 
+  record TeamMemberRole {
+    UID userID;
+    string username;
+    FullName fullName;
+    TeamRole role;
+  }
+
+  record TeamInfo {
```
Comment (zapu) [inline protocol/avdl/keybase1/teams.avdl]: This naming may be too generic if it's only used for untrusted team info.

Comment (AMarcedone) [reply protocol/avdl/keybase1/teams.avdl]: I don't feel strongly about it, but being more generic makes it more likely to be reused. I think the fact that the rpc name has untrusted in the name is enough. Happy to change it if you think it's better.

File: go/teams/teams.go [hunk 3]
```diff
@@ -2707,3 +2707,69 @@ func KeySummary(t Teamer) string {
...
+		Description:   resp.Description,
+		InTeam:        resp.InTeam,
+		NumMembers:    resp.NumMembers,
+		Open:          resp.Open,
+		PublicAdmins:  resp.PublicAdmins,
+		PublicMembers: make(map[keybase1.UID]keybase1.TeamMemberRole),
+	}
+
+	for _, uc := range resp.PublicUserCards {
+		teamInfo.PublicMembers[uc.UID] = keybase1.TeamMemberRole{UserID: uc.UID, FullName: keybase1.FullName(uc.FullName)}
```
Comment (buoyad) [inline go/teams/teams.go]: Are admins included in this list? We'll need fullnames for them too.

File: protocol/avdl/keybase1/teams.avdl [hunk 4]
```diff
@@ -185,6 +185,25 @@ protocol teams {
...
+    TeamName name;
+    boolean inTeam; 
+    boolean open;
+    string description;
+    array<string> publicAdmins;
+    int numMembers;
+    map<UID,TeamMemberRole> publicMembers;
```
Comment (buoyad) [inline protocol/avdl/keybase1/teams.avdl]: `[]TeamMemberRole` is better for this.

File: go/teams/teams.go (hunk 3 above)
Comment (AMarcedone) [reply go/teams/teams.go]: The public ones are I think.

**Comment by AMarcedone** [comment]
Updated. It now depends on this server change https://github.com/keybase/keybase/pull/5148

**Pull request merged**